Geometry/MTDCommonData/src/ETLNumberingScheme.cc

```diff
@@ -58,21 +73,21 @@ uint32_t ETLNumberingScheme::getUnitID(const MTDBaseNumber& baseNumber) const {
 
   // error checking
 
-  if (modtyp != 0) {
+  if ((modtyp != 0 && preTDR) || (modtyp == 0 && !preTDR)) {
     edm::LogWarning("MTDGeom") << "ETLNumberingScheme::getUnitID(): "
                                << "****************** Bad module name = " << modtyp
                                << ", Volume Name = " << baseNumber.getLevelName(4);
     return 0;
   }
 
-  if (1 > modCopy || 176 < modCopy) {
+  if ((preTDR && (1 > modCopy || 176 < modCopy)) || (!preTDR && (1 > modCopy || 292 < modCopy))) {
```

Comment:
magic number 176?

Reply:
@kpedro88 the meaning is pretty obvious and this limit is used nowhere else at present, anyway in view of a possible future differentiation of other parts of code I have added some constants to ETLDetId